src/fec_base.h

```diff
@@ -746,9 +778,72 @@ void FecCode<T>::decode_Lagrange(
         S.set(i, N_p.eval(inv_r_powers->get(i + 1)));
     }
     S.neg();
-    // std::cout << "S:"; S.dump();
     S.mul(&A, k - 1);
-    // std::cout << "S x A:"; S.dump();
+    // No need to mod x^n since only last n_data coefs are obtained
+    // output is n_data length
+    for (unsigned i = 0; i < this->n_data; i++)
+        output->set(i, S.get(i));
+}
+
+template <typename T>
+void FecCode<T>::decode_vec_lagrange(
+    vec::Vector<T>* output,
+    const std::vector<Properties>& props,
+    off_t offset,
+    vec::Vector<T>* fragments_ids,
+    vec::Vector<T>* words,
+    vec::Vector<T>* vx,
+    int vx_zero)
+{
+    if (this->fft == nullptr) {
+        throw LogicError("FEC base: FFT must be initialized");
+    }
+    if (this->fft_full == nullptr) {
+        throw LogicError("FEC base: FFT must be initialized");
```

Comment:
Same error message as above, but difference root cause: that can be misleading (could be "FEC base: FFT full must be initialized").